cfme/tests/intelligence/reports/test_reports.py

```diff
@@ -290,3 +297,59 @@ def test_send_text_custom_report_with_long_condition(
     )
     # assert that the pattern was not found in the logs
     assert log.validate(), "Found error message in the logs."
+
+
+@pytest.mark.tier(2)
+@pytest.mark.meta(automates=[1504010])
+@pytest.mark.provider([BaseProvider], selector=ONE_PER_CATEGORY)
+def test_reports_online_vms(appliance, setup_provider, provider, request, vm):
+    """
+    Polarion:
+        assignee: pvala
+        casecomponent: Reporting
+        caseimportance: medium
+        initialEstimate: 1/2h
+        testSteps:
+            1. Add a provider.
+            2. Power off a VM.
+            3. Queue report (Operations > Virtual Machines > Online VMs (Powered On)).
+            4. See if the powered off VM is present in the queued report.
+        expectedResults:
+            1.
+            2.
+            3.
+            4. VM must not be present in the report data.
+
+    Bugzilla:
+        1504010
+    """
+    subject_vm = appliance.rest_api.collections.vms.get(name=vm)
+    response = subject_vm.action.stop()
+    assert_response(appliance)
+    # Wait for task to be finished.
+    wait_for(
+        lambda: response.task.state == "Finished",
+        fail_func=response.task.reload,
+        timeout=50,
+        delay=2,
+    )
+    # Wait for VM power state to change
+    # Note: Simply waiting for vm power_state to change doesn't work
+    subject_vm.reload()
+    wait_for(
```

Comment:
Q: Can you use a method here? Something like `wait_for_vm_power_state`, I see something like this in `cfme.tests.service.test_rest_services`. Could that be a common VM method? (Maybe out of scope for this PR)

https://github.com/ManageIQ/integration_tests/blob/861426751dd237366a91410fa0b76170d0351eb2/cfme/tests/services/test_rest_services.py#L57-L62

Reply:
I could add this method to VM's BaseEntity, but Idk why for some reason, when I simply checked VM's power state(like it does in `wait_for_vm_power_state`) my test kept failing, but when I used the current implementation, it worked well. I'll have to investigate it more I guess.

Reply:
Hey @john-dupuy, so I changed the num_sec and delay, and it seems to be working fine. I'll raise a separate PR for adding `wait_for_vm_power_state` to the BaseEntity, thanks for the idea though, it's good! :)

Reply:
Awesome, thanks @valaparthvi!